### ISSUE TITLE
Same QR code when multiple shown in a single page

### ISSUE THREAD

**Issue by fayazara** [kozakdenys/qr-code-styling#72]
I am seeing the same QR Code when I try to show multiple in a single page, I am using this in a Vue project, anyone knows how to fix this.

**Comment by VictoriaTheBrave** [comment]
I forked and fixed this issue by myself: the problem is in ids and URLs of svg. For different svg-s there are the same names and thus the same paths and colors are applied to every QR on the page
Here is the example of an updated line:
`this._dotsClipPath.setAttribute("id", `${this._svgPrefix}-clip-path-dot-color`);`
E.g. I've added `this._svgPrefix` which is unique for every QR code

**Comment by chiptu** [comment]
Thanks a lot for your response @VictoriaTheBrave .

In my case i switched to canvas type,  if you're using vuejs you can just change type in option data from 
```
const options = {
            type: 'svg'
            ... }
```
to
```
const options = {
            type: 'canvas'
            ... }
```

**Comment by khosiDigital** [comment]
> I forked and fixed this issue by myself: the problem is in ids and URLs of svg. For different svg-s there are the same names and thus the same paths and colors are applied to every QR on the page Here is the example of an updated line: `this._dotsClipPath.setAttribute("id", `${this._svgPrefix}-clip-path-dot-color`);` E.g. I've added `this._svgPrefix` which is unique for every QR code
> 
> @kodmunki Maybe it could help you

Hey @VictoriaTheBrave Im having the same issue right now with SVG, have you made a pull request so that this can be included in the master branch! That would help us all so much!

**Comment by khosiDigital** [comment]
I managed to get this to work. Its @VictoriaTheBraves answer but what was missing in the answer was passing the new clip path id into the create colour function. 
in QRSVG.ts line 174

`   //create a random number and then create a new clip path Id
    const svgId = Math.random().toString();
    this._dotsClipPath.setAttribute("id", "clip-path-" + svgId);

   // pass the random number into the create colour function. 
    this._createColor({
      options: options.dotsOptions?.gradient,
      color: options.dotsOptions.color,
      additionalRotation: 0,
      x: xBeginning,
      y: yBeginning,
      height: count * dotSize,
      width: count * dotSize,
      name: svgId
    });`

Im going to create a merge request, hopefully  this gets commited because this is A MAJOR BUG

**Comment by BrassStack** [comment]
I was able to apply the changes to `QRSVG.ts` from the Solana Labs fork's commit `e9c374c` to a clone of this one to fix this issue.

**Comment by kozakdenys** [comment]
Hi @BrassStack @kodmunki @fedyk @khosiDigital @fayazara @chiptu @VictoriaTheBrave , the issue with 2 qr-codes on the same page was fixed in version [v1.7.1](https://github.com/kozakdenys/qr-code-styling/releases/tag/v1.7.1) and above.
Sorry for missing for so long 🥲
